terraform/modules/quorum-cluster-region/quorum-cluster-region-backup-event.tf

```diff
@@ -0,0 +1,399 @@
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  type        = "zip"
+  source_file  = "${var.BackupLambda_source_file}"
+  output_path = "${var.aws_region}-${var.BackupLambda_output_path}"
+  depends_on = ["null_resource.BackupLambda"]
+}
+
+resource "aws_kms_key" "a" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  description = "Used for encrypting SSH keys on S3"
+  tags {
+     name = "BackupLambda-${var.network_id}-${var.aws_region}-KMS"
+  }
+}
+
+output "aws_kms_key_test" {
+  value = "${aws_kms_key.a.*.id}"
+}
+
+# Encrypt the contents of the file located at var.private_key_path
+data "aws_kms_ciphertext" "EncryptSSHKey" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  key_id = "${aws_kms_key.a.id}"  
+  plaintext = "${file("${var.private_key_path}")}"
+}
+
+# Save the encrypted contents to the file specified at filename
+resource "local_file" "EncryptedSSHKey" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  content = "${base64decode("${data.aws_kms_ciphertext.EncryptSSHKey.ciphertext_blob}")}"
+  filename = "${var.enc_ssh_path}-${var.aws_region}"
+}
+
+resource "aws_kms_grant" "a" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  name              = "kms-grant-${var.network_id}-${var.aws_region}"
+  key_id            = "${aws_kms_key.a.key_id}"
+  grantee_principal = "${aws_iam_role.iam_for_BackupLambda.arn}"
+  operations        = ["Encrypt", "Decrypt"]
+}
+
+output "aws_s3_bucket" {
+  value = "${element(concat(aws_s3_bucket.quorum_backup.*.id, list("")), 0)}"
+}
+
+resource "aws_security_group" "allow_all" {
+    count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+    name        = "BackupLambdaSSH-${var.network_id}-${var.aws_region}-allow_all"
+    description = "Allow all outgoing traffic"
+    vpc_id = "${aws_vpc.quorum_cluster.id}"
+
+    egress {
+        from_port   = 0
+        to_port     = 0
+        protocol    = "-1"
+        cidr_blocks = ["0.0.0.0/0"]
+        description = "Allow all traffic"
+    }
+    ingress {
+      from_port = 22
+      to_port = 22
+      protocol = "tcp"
+      cidr_blocks = ["0.0.0.0/0"]
+      description = "Allow all SSH traffic"
+    }
+  tags {
+     name = "BackupLambda-${var.network_id}-${var.aws_region}-SG"
+  }
+}
+
+// use the next value after data.template_file.quorum_observer_cidr_block
+data "template_file" "quorum_maker_cidr_block_lambda" {
+  template = "$${cidr_block}"
+
+  vars {
+    cidr_block = "${cidrsubnet(data.template_file.quorum_cidr_block.rendered, 2, 3)}"
+  }
+}
+
+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical
+}
+
+resource "aws_instance" "web1" {
```

Comment:
More descriptive name please. I don't actually know what this instance does.

Reply:
The entire Terraform configuration was developed modularly, in isolation from each other. This was used during development to ensure that instances can be reached and connected via SSH, to test VPC, and to test the BackupLambda subnet.